provider/marathon/marathon_test.go

```diff
@@ -50,27 +51,12 @@ func TestMarathonLoadConfigNonAPIErrors(t *testing.T) {
 		expectedBackends  map[string]*types.Backend
 	}{
 		{
-			desc: "simple application",
-			application: marathon.Application{
-				Ports:  []int{80},
-				Labels: &map[string]string{},
-			},
-			task: marathon.Task{
-				Host:  "localhost",
-				Ports: []int{80},
-				IPAddresses: []*marathon.IPAddress{
-					{
-						IPAddress: "127.0.0.1",
-						Protocol:  "tcp",
-					},
-				},
-			},
+			desc:        "simple application",
+			application: application(appPorts(80)),
```

Comment:
I should already only be setting it when I actually need it. (There are a number of `application()`-only invocations where I don't pass the `appPorts` builder function.)

I thought about setting default ports on the app and task but dropped the idea eventually: We have a lot of different cases where we sometimes need just the Marathon port and sometimes the task port; sometimes they need to be equal and sometimes they don't.

So if I added a default, I'd also need to override it explicitly sometimes. In the end, this was too intransparent for me, so I went down the slightly more explicit and but also more clear route.